aten/src/ATen/native/composite/soft_margin_loss.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+#include <ATen/core/Reduction.h>
+#include <ATen/core/TensorBody.h>
+
+namespace at {
+namespace native {
+
+template <typename OPS>
+Tensor& soft_margin_loss_out(
+    const Tensor& input,
+    const Tensor& target,
+    int64_t reduction,
+    Tensor& output) {
+  // compute inplace variant of: output = at::log(1. + at::exp(-input *
+  // target));
+  OPS::neg_out(output, input);
```

Comment:
Doesn't this throw a resize warning when the `output` is the correct ahpw, that is, of shape `{}` whenever `reduction != None`? I believe this should be implemented like:
```c++
TORCH_CHECK(input.sizes() == target.sizes(), "...");
at::native::resize_output(output, output_shape);
auto x = reduction == Reduction::None ? output : at::empty_like(input);
```
and then just use `result` at the end via `mean_out` and`sum_out` if `reduction != None`.

Reply:
I think you are right.
Maybe this (and your other suggestion) might be better done in another PR.